ConnectSecure/Deploy-ConnectSecure-V4.ps1

```diff
@@ -95,6 +96,12 @@ if([string]::IsNullOrEmpty($TenantID) -or [string]::IsNullOrWhiteSpace($TenantID
     $TenantID = '000000000000000000'    # Numeric Tenant ID string from the deployment script
 }
 
+# Manually set the user secret for CyberCNS V4 instance, if not passed in via variable or parameter already:
+if([string]::IsNullOrEmpty($userSecret) -or [string]::IsNullOrWhiteSpace($userSecret)) {
+    $userSecret = '000000000000000000'    # Numeric User secret string from the deployment script
```

Comment:
I'm not the biggest fan of including this inside the script itself but in this case given the relative lack of importance of this new field for most MSP use cases, it's probably fine.

However, as a rule, leaving a placeholder value that can be set to a real value when deployed, does make it easier to accidentally enter a real value while editing/deploying and without realizing, commit back to git, potentially sharing a non-generic value in a public repository. But this is balanced with this value potentially not being worth adding another custom field for and hardcoding into the script for ease of use, as its only value is tracking the ConnectSecure user that generated the installer, which will likely be a generic or primary admin for MSP, and certainly is not terribly helpful to track when added to a generic deployment script in an RMM.

All that to say this is probably fine as-is and I might make the same choice, so for now it seems fine to leave it even if suboptimal in an ideal world.

Reply:
I also made it a script parameter, and I agree that hardcoding it probably isn't the best idea. Since the parameter can be set as default in Ninja, there's probably no value in leaving this. I will remove it.